cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt

```diff
@@ -102,27 +105,60 @@ private class NodeRunner(baseDebugPort: Int, baseMonitoringPort: Int) {
+
+        val createTempScriptFile = {
+            val tmpFile: File = File.createTempFile("corda_node_script_launcher", "")
+            BufferedWriter(OutputStreamWriter(FileOutputStream(tmpFile))).use {
+                //Start shell to keep window open unless java terminated normally or due to SIGTERM:
+                it.write("""
+                        #!/bin/sh
+                        set -v
+                        unlink $tmpFile
+                        cd "$workingDir"
+                        command -v rlwrap
+                        if [ -${'$'}? -eq 0 ]
+                        then
+                            rlwrap -H ~/.crash_history --always-readline ${command.joinToString(" ")}; [ ${'$'}? -eq 0 -o ${'$'}? -eq 143 ] || sh
+                        else
+                            ${command.joinToString(" ")}; [ ${'$'}? -eq 0 -o ${'$'}? -eq 143 ] || sh
+                        fi
+                    """.trimIndent())
+            }
+            tmpFile.setExecutable(true)
+            tmpFile
+        }
         val params = when (os) {
             OS.MACOS -> {
-                listOf("osascript", "-e", """tell app "Terminal"
-    activate
-    delay 0.5
-    tell app "System Events" to tell process "Terminal" to keystroke "t" using command down
-    delay 0.5
-    do script "bash -c 'cd \"$workingDir\" ; \"${command.joinToString("""\" \"""")}\" && exit'" in selected tab of the front window
-end tell""")
+                val tmpFile = createTempScriptFile()
+                listOf("osascript", "-e", """
+                    tell app "Terminal"
+                        activate
+                        delay 0.5
+                        tell app "System Events" to tell process "Terminal" to keystroke "t" using command down
+                        delay 0.5
+                        do script "bash \"$tmpFile\" && exit" in selected tab of the front window
+                    end tell
+                """.trimIndent())
             }
             OS.WINDOWS -> {
                 listOf("cmd", "/C", "start ${command.joinToString(" ") { windowsSpaceEscape(it) }}")
             }
             OS.LINUX -> {
-                // Start shell to keep window open unless java terminated normally or due to SIGTERM:
-                val unixCommand = "${unixCommand(command)}; [ $? -eq 0 -o $? -eq 143 ] || sh"
-                if (isTmux()) {
-                    listOf("tmux", "new-window", "-n", nodeName, unixCommand)
-                } else {
-                    listOf("xterm", "-T", nodeName, "-e", unixCommand)
-                }
+                val tmpFile = createTempScriptFile()
+                val shellScript = """
+                    NODE_NAME='$nodeName'
+                    COMMAND="$tmpFile"
+                    if [ -n ${'$'}{CORDA_NODE_RUNNER_TERMINAL_COMMAND+x} ]
+                    then
+                        eval ${'$'}CORDA_NODE_RUNNER_TERMINAL_COMMAND
```

Comment:
I forgot to mention that it looks for the `$NODE_NAME` and `$COMMAND` placeholders in the value of the environmental variable `CORDA_NODE_RUNNER_TERMINAL_COMMAND`, e.g. I use `tilix` as a terminal emulator and I have
```
[woggioni@19DUB-DEL05 ~]$ echo $CORDA_NODE_RUNNER_TERMINAL_COMMAND
tilix -t "$NODE_NAME" -e "$COMMAND"
```